JustSaying.TestingFramework/TestEnvironment.cs

```diff
@@ -37,12 +37,24 @@ public static RegionEndpoint Region
         /// <summary>
         /// Gets a value indicating whether an AWS simulator is configured for use.
         /// </summary>
-        public static bool IsSimulatorConfigured => !string.IsNullOrEmpty(SimulatorUrl);
+        public static bool IsSimulatorConfigured => (SimulatorUrl != null);
 
         /// <summary>
         /// Gets the URL for the configured AWS simulator, if any.
         /// </summary>
-        public static string SimulatorUrl => Environment.GetEnvironmentVariable("AWS_SERVICE_URL") ?? string.Empty;
+        public static Uri SimulatorUrl
+        {
+            get
+            {
+                var awsEnv = Environment.GetEnvironmentVariable("AWS_SERVICE_URL");
+                if (string.IsNullOrWhiteSpace(awsEnv))
+                {
+                    return null;
+                }
+
+                return new Uri(awsEnv);
```

Comment:
Suggest using the constructor that specifies `UriKind.Absolute`.

Reply:
Is that done to prevent invalid data? Should it also be done on other code that calls `new Uri` ? or is it just a concern in this case since Env vars are more prone to bad/missing data?

Reply:
Basically yes, it flushes out misconfiguration earlier than at the point it gets consumed by something trying to do a request. I do it most times I create a `Uri` it as it's very rare I'd ever need a relative URI. Depends on context obviously, but in this case I can't see any reason why you'd be pointing at a relative URI for an AWS resource.